be/src/vec/data_types/data_type_map.cpp

```diff
@@ -0,0 +1,157 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "data_type_map.h"
+
+#include "gen_cpp/data.pb.h"
+#include "vec/data_types/data_type_factory.hpp"
+#include "vec/columns/column_array.h"
+#include "vec/columns/column_map.h"
+#include "vec/common/assert_cast.h"
+
+namespace doris::vectorized {
+
+DataTypeMap::DataTypeMap(const DataTypePtr& keys_, const DataTypePtr& values_)
+{
+    key_type = keys_;
+    value_type = values_;
+
+    keys = std::make_shared<DataTypeArray>(key_type);
+    values = std::make_shared<DataTypeArray>(value_type);
+}
+
+std::string DataTypeMap::to_string(const IColumn& column, size_t row_num) const {
+    const ColumnMap & map_column = assert_cast<const ColumnMap &>(column);
+    const ColumnArray::Offsets64& offsets = map_column.get_offsets();
+
+    size_t offset = offsets[row_num - 1];
+    size_t next_offset = offsets[row_num];
+
+    const IColumn & nested_keys = map_column.get_keys();
+    const IColumn & nested_values = map_column.get_values();
+
+    std::stringstream ss;
+    ss << "{";
+    for (size_t i = offset; i < next_offset; ++i)
+    {
+        if (i != offset)
+            ss << ", ";
+        ss << "'" << keys->to_string(nested_keys, i);
```

Comment:
'' is only needed for key of type string

Reply:
done!